src/vm/methodtablebuilder.cpp

```diff
@@ -1517,12 +1518,15 @@ MethodTableBuilder::BuildMethodTableThrowing(
         if (hr == S_OK && (strcmp(nameSpace, "System.Runtime.Intrinsics.X86") == 0))
 #endif
         {
-            if (IsCompilationProcess())
+#ifdef CROSSGEN_COMPILE
+            if (!IsNgenPDBCompilationProcess() &&
+                GetAppDomain()->ToCompilationDomain()->GetTargetModule() != g_pObjectClass->GetModule())
```

Comment:
Does this check mean that we only do this for corelib?

Reply:
(That is, we only support HWIntrinsics for corelib and we say "disallowed" for everything else).

Reply:
The comment below needs editing - I think it's actually incorrect now. Also, it would be good to clarify exactly what the above condition is checking.